src/backend/utils/init/miscinit.c

```diff
@@ -1688,6 +1688,17 @@ process_shared_preload_libraries(void)
 {
 	process_shared_preload_libraries_in_progress = true;
 
+	if ((shared_preload_libraries_string == NULL || strlen(shared_preload_libraries_string) == 0)
+		&& (IsBootstrapProcessingMode() || IsInitProcessingMode()))
```

Comment:
just curious, for bootstrap and init, the `shared_preload_libraries_string` should always be null or empty.
probably an assertion is better. No strong opinions on this.